packages/core/src/reducers/core.ts

```diff
@@ -145,7 +145,11 @@ export const coreReducer = (
         };
       } else {
         const oldData: any = _.get(state.data, action.path);
-        const newData = action.updater(oldData);
+        let newData = action.updater(oldData);
+        if (newData === '' && newData.constructor.name === 'String') {
```

Comment:
do we need the `newData.constructor.name === 'String' ` check?

Reply:
I don't think so

Reply:
It seems like it's not needed, since only strings and dates produce an empty string when cleared and the data type for dates is string anyway. I pushed an updated commit.